src/app/users/users.component.ts

```diff
@@ -102,12 +101,15 @@ export class UsersComponent implements OnInit, AfterViewInit {
     });
   }
 
-  deleteRole(user: any, index: number) {
+  deleteRole(user: any, index: number, position: number) {
     // Make copy of user so UI doesn't change until DB change succeeds
-    const tempUser = { ...user, roles: [ ...user.roles ] };
+    let tempUser = user;

```

Comment:
Please keep this the same.  We need to create a deep copy of the user object so the view does not get updated before the HTTP request to CouchDB is successful.